core/src/main/java/io/undertow/server/protocol/framed/AbstractFramedChannel.java

```diff
@@ -933,8 +934,9 @@ void notifyFrameReadComplete(AbstractFramedStreamSourceChannel<C, R, S> channel)
         @Override
         public void handleEvent(final StreamSourceChannel channel) {
             //clear the task queue before reading
-            while (!taskRunQueue.isEmpty()) {
-                taskRunQueue.poll().run();
+            Runnable runnable;
+            while ((runnable = taskRunQueue.poll()) != null) {
+                runnable.run();
```

Comment:
Looks like there's one more instance of this pattern on `newFrames` in `void flushSenders()`

Reply:
Good catch! I created https://issues.jboss.org/browse/UNDERTOW-1621 and I'll add a fix for this in this PR.

Reply:
Thanks!